From 41bee2100a9b7905f82022aa2ff0b72aeadd1116 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.king@canonical.com>
Date: Mon, 10 Mar 2014 10:23:49 +0000
Subject: [PATCH 20/28] cthd_engine: remove redundant reads of thermal zone
 type files

takeover_thermal_control() and giveup_thermal_control() both
contain redundant reads of the thermal zone type fields, so remove
this code.

Signed-off-by: Colin Ian King <colin.king@canonical.com>
---
 src/thd_engine.cpp | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/thd_engine.cpp b/src/thd_engine.cpp
index 565b3d4..c8beed4 100644
--- a/src/thd_engine.cpp
+++ b/src/thd_engine.cpp
@@ -444,14 +444,8 @@ void cthd_engine::takeover_thermal_control() {
 
 				i = atoi(entry->d_name + strlen("thermal_zone"));
 				std::stringstream policy;
-				std::stringstream type;
-				std::string type_val;
 				std::string curr_policy;
 
-				type << "thermal_zone" << i << "/type";
-				if (sysfs.exists(type.str().c_str())) {
-					sysfs.read(type.str(), type_val);
-				}
 				policy << "thermal_zone" << i << "/policy";
 				if (sysfs.exists(policy.str().c_str())) {
 					sysfs.read(policy.str(), curr_policy);
@@ -487,13 +481,7 @@ void cthd_engine::giveup_thermal_control() {
 
 				i = atoi(entry->d_name + strlen("thermal_zone"));
 				std::stringstream policy;
-				std::stringstream type;
-				std::string type_val;
 
-				type << "thermal_zone" << i << "/type";
-				if (sysfs.exists(type.str().c_str())) {
-					sysfs.read(type.str(), type_val);
-				}
 				policy << "thermal_zone" << i << "/policy";
 				if (sysfs.exists(policy.str().c_str())) {
 					sysfs.write(policy.str(), zone_preferences[cnt++]);
-- 
1.9.0

